entry.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+setup() {
+    # echo commands to the terminal output
+    set -ex
+
+    # Check whether there is a passwd entry for the container UID
+    myuid=$(id -u)
+    mygid=$(id -g)
+    # turn off -e for getent because it will return error code in anonymous uid case
+    set +e
+    uidentry=$(getent passwd $myuid)
+    set -e
+
+    # If there is no passwd entry for the container UID, attempt to create one
+    if [ -z "$uidentry" ] ; then
+        if [ -w /etc/passwd ] ; then
+            echo "$myuid:x:$myuid:$mygid:${SPARK_USER_NAME:-anonymous uid}:$SPARK_HOME:/bin/false" >> /etc/passwd
+        else
+            echo "Container ENTRYPOINT failed to add passwd entry for anonymous UID"
+        fi
+    fi
+
+    if [ -z "$JAVA_HOME" ]; then
+        JAVA_HOME=$(java -XshowSettings:properties -version 2>&1 > /dev/null | grep 'java.home' | awk '{print $3}')
+    fi
+
+    SPARK_CLASSPATH="$SPARK_CLASSPATH:${SPARK_HOME}/jars/*"
+    env | grep SPARK_JAVA_OPT_ | sort -t_ -k4 -n | sed 's/[^=]*=\(.*\)/\1/g' > /tmp/java_opts.txt
+    readarray -t SPARK_EXECUTOR_JAVA_OPTS < /tmp/java_opts.txt
+
+    if [ -n "$SPARK_EXTRA_CLASSPATH" ]; then
+        SPARK_CLASSPATH="$SPARK_CLASSPATH:$SPARK_EXTRA_CLASSPATH"
+    fi
+
+    if ! [ -z ${PYSPARK_PYTHON+x} ]; then
+        export PYSPARK_PYTHON
+    fi  
```

Comment:
1) please, use `if [ -n ...` everywhere you are using `if ! [ -z ...`

2) I think it does not what you expect:
```
13:04:49 ✔ taurus:~$ TEST="" ; echo ${TEST+x}
x
13:04:53 ✔ taurus:~$ TEST="AAA" ; echo ${TEST+x}
x
```

So, effectively condition is always true. Isn't it?
It can be simplified to:

```suggestion
    [ -n "${PYSPARK_PYTHON:-}" ] && export "${PYSPARK_PYTHON}
```

There are several such places below you can fix and simplify.